SofaKernel/modules/SofaBaseUtils/AddResourceRepository.cpp

```diff
@@ -68,21 +73,39 @@ void BaseAddResourceRepository::parse(sofa::core::objectmodel::BaseObjectDescrip
     if (!sofa::helper::system::FileSystem::exists(tmpAddedPath))
     {
         msg_error(this) << tmpAddedPath + " does not exist !";
-        return;
+        return false;
     }
     //third, check if it is really a directory
     if (!sofa::helper::system::FileSystem::isDirectory(tmpAddedPath))
     {
         msg_error(this) << tmpAddedPath + " is not a valid directory !";
-        return;
+        return false;
     }
 
+    if (!m_currentAddedPath.empty())
+        m_repository->removePath(m_currentAddedPath);
+
     m_currentAddedPath = FileSystem::cleanPath(tmpAddedPath);
+    if (m_currentAddedPath != d_repositoryPath.getValue())
+        d_repositoryPath.setValue(m_currentAddedPath);
+
     m_repository->addLastPath(m_currentAddedPath);
     msg_info(this) << "Added path: " << m_currentAddedPath;
 
     if(this->f_printLog.getValue())
         m_repository->print();
+    return true;
+}
+
+
+void BaseAddResourceRepository::parse(sofa::core::objectmodel::BaseObjectDescription* arg)
+{
+    Inherit1::parse(arg);
+
+    if (parse())
+        m_componentstate.setValue(sofa::core::objectmodel::ComponentState::Valid);
```

Comment:
SetValue on componentState should not be required: `bool parse()` calls getValue on the m_repositoryPath, which in turn will call the updateCallback, which himself sets the component State. Thus setting it again would be useless